planning/mission_planner/CMakeLists.txt

```diff
@@ -4,24 +4,24 @@ project(mission_planner)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
-ament_auto_add_library(mission_planner_node SHARED
-  lib/mission_planner_base.cpp
+ament_auto_add_library(${PROJECT_NAME} SHARED
+  src/mission_planner/mission_planner.cpp
   src/mission_planner_lanelet2/mission_planner_lanelet2.cpp
   src/mission_planner_lanelet2/utility_functions.cpp
 )
 
-rclcpp_components_register_node(mission_planner_node
+rclcpp_components_register_node(${PROJECT_NAME}
   PLUGIN "mission_planner::MissionPlannerLanelet2"
-  EXECUTABLE mission_planner
+  EXECUTABLE mission_planner_node
 )
 
-ament_auto_add_library(goal_pose_visualizer_node SHARED
+ament_auto_add_library(goal_pose_visualizer SHARED
   src/goal_pose_visualizer/goal_pose_visualizer.cpp
 )
 
-rclcpp_components_register_node(goal_pose_visualizer_node
+rclcpp_components_register_node(goal_pose_visualizer
   PLUGIN "mission_planner::GoalPoseVisualizer"
-  EXECUTABLE goal_pose_visualizer
+  EXECUTABLE goal_pose_visualizer_node
```

Comment:
The 1st argument of rclcpp_components_register_node is library (`.so` file). So this is not a node by itself. EXECUTABLE is a node that can be launch by itself, so I think this name is natural.

Reply:
Right, it's not a node, but it can be called a "node component".
So I don't feel it's unnatural.

Also, regarding EXECUTABLE, I guess users don't want to add `_node` whenever they launch a node.
It's because they want to launch an application, not a node itself.

Therefore, I recommend reverting these changes.
But you can ask other developers for other opinions.

Or, if you can show some references, I can approve these changes.

Reply:
> Right, it's not a node, but it can be called a "node component".
So I don't feel it's unnatural.

I feel very unnatural to call this a node. A composable node is a class within the library, not the library itself. A library can have multiple composable nodes.

> Also, regarding EXECUTABLE, I guess users don't want to add _node whenever they launch a node.
It's because they want to launch an application, not a node itself.

I understand. Remove the `_node` from the EXECUTABLE. But I would like to think of another name for the library.

Reply:
I counted the naming in Autoware. It's almost the same, but there are a few more packages with a suffix of `_node`. Focusing on EXECUTABLE, there are 48 with suffix and 30 without suffix.

| library name | executable name | packages |
| ---- | ---- | ---- |
| name | name_node | 30 |
| name_node | name | 26 |
| name_component | name_node | 6 |
| name_nodelet | name_node | 5 |
| name | name_exe | 3 |
| name_node | name_exe | 3 |
| name_lib | name | 2 |
| name_core | name_node | 1 |
| name_core | name | 1 |
| name_component | name | 1 |

Reply:
I think `{name}_component` is a good alternative naming.

Reply:
Done https://github.com/autowarefoundation/autoware.universe/pull/1561/commits/bbad49987fc5651b13dd9c863d314e4e13123fff